### DISCUSSION TITLE
Why was the search function removed

### DISCUSSION THREAD

**Discussion by AdalaceV2** [brianpetro/obsidian-smart-connections#593] [Q&A]
My favorite use of this plugin is to do semantic searches, and now it seems to be gone. Any reason why?

Thank you again for the work you're doing :)

**Comment by brianpetro** [comment]
Hi @AdalaceV2 

It was a few things, but the straw that broke the camel's back was that it was suggested that it was causing Obsidian to crash in https://github.com/brianpetro/obsidian-smart-connections/issues/583#issuecomment-2097022842

While I couldn't reproduce the issue myself, the "search" feature was already confusing/frustrating for many people who expected it to work like a traditional keyword search.

And the input UI/UX also needed some work.

For those reasons, I thought it would be best to disable it until I have time to address its issues and reimplement it in a more intuitive way.

🌴

**Reply by AdalaceV2** [reply]
Hi @brianpetro,

Thank you for your insights on the search feature issues. I agree that the current implementation has been causing some frustration, especially with the crashes.

However, the search feature is incredibly useful and adds significant value to Obsidian. Can we modify the trigger mechanism for the new implementation? Specifically, it might be more user-friendly if the search only starts after the user presses the return key. This change could mitigate the problem of the search kicking off prematurely and enhance the overall user experience.

Looking forward to the updated version!

**Comment by abhuva** [comment]
Wow, i am glad i stumbled upon this. I am also making mainly use of the search function - seems no updates for me till this is solved =)
While i agree that the UI/UX needs some work for this, i cant understand how this can be frustrating for people - its just a different kind of search (if i want keyword search i use Omnisearch or similar stuff). If this is mainly about expectations - it might be easy solvable by having some kind of mini introduction available that explains embeddings.

I hope this functionality makes it way back rather sooner than later.

**Comment by brianpetro** [comment]
👀

![Screenshot 2024-05-15 at 5 50 50 PM](https://github.com/brianpetro/obsidian-smart-connections/assets/1886014/3885155d-7275-43a6-b409-d1e38754357f)

**Reply by arminta7** [reply]
Yay! Been missing this.

**Reply by AdalaceV2** [reply]
Superb!

**Comment by AdalaceV2** [comment]
Do you have an estimate as to when 2.2 will be generally available?
